### PR TITLE
TS-19-Sprint-preview

### PR DESCRIPTION
**Pull request by Niknikson**

### CODE REVIEW

File: frontend/src/actions/lesson.js [hunk 1]
```diff
@@ -1,95 +1,64 @@
...
-export const selectGroupId = (res) => {
-    return {
-        type: actionTypes.SELECT_GROUP_ID,
-        result: res,
-    };
-};
+export const selectGroupId = (result) => ({
+    type: actionTypes.SELECT_GROUP_ID,
+    result,
+});
```
Comment (Niknikson) [inline frontend/src/actions/lesson.js:59]: maybe to use payload or id

File: frontend/src/components/Semester/SemesterList/SemesterList.js [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+            ? getGroupsOptionsForSelect(semester.semester_groups)
+            : [];
+        const finishGroups = [...semesterGroupsOptions];
+        if (isEqual(beginGroups, finishGroups)) {
+            setOpenSuccessSnackbar(
+                i18n.t(GROUP_EXIST_IN_THIS_SEMESTER, {
+                    cardType: i18n.t(COMMON_GROUP_TITLE),
+                    actionType: i18n.t(EXIST_LABEL),
+                }),
+            );
```
Comment (Niknikson) [inline frontend/src/components/Semester/SemesterList/SemesterList.js]: move to saga

**Review by HiroFive — APPROVED** [review]

File: frontend/src/actions/schedule.js [hunk 3]
```diff
@@ -123,30 +121,38 @@ export const checkAvailabilityScheduleStart = (item) => {
...
     };
 };
-
-export const setGroupSchedule = (result) => {
+export const setItemGroupId = (res) => {
     return {
-        type: actionTypes.SET_GROUP_SCHEDULE,
-        result,
+        type: actionTypes.SET_ITEM_GROUP_ID,
+        result: res,
```
Comment (ViktoriiaKosovan) [inline frontend/src/actions/schedule.js]: Could you use one style of payload in actions?

File: frontend/src/actions/schedule.js [hunk 4]
```diff
@@ -183,49 +191,59 @@ export const setScheduleType = (result) => {
...
     };
 };
 
-export const setSemesterList = (result) => {
+export const setTeacherViewType = (type) => {
     return {
-        type: actionTypes.SET_SEMESTER_LIST,
-        result,
+        type: actionTypes.SET_TEACHER_VIEW_TYPE,
+        result: type,
```
Comment (ViktoriiaKosovan) [inline frontend/src/actions/schedule.js]: as above

File: frontend/src/components/EditCurrentSchedule/ScheduleBoard/ScheduleBoardItem.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    const {
+        deleteScheduleItem,
+        checkRoomAvailability,
+        itemData,
+        getLessonsByGroupId,
+        selectByGroupId,
+        t,
+        openDialogWithData,
+    } = props;
+    const [anchorEl, setAnchorEl] = React.useState(null);
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/EditCurrentSchedule/ScheduleBoard/ScheduleBoardItem.js]: @HiroFive Could you use only useState without React?

File: frontend/src/components/EditCurrentSchedule/ScheduleBoard/ScheduleBoardItem.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        setAnchorEl(event.currentTarget);
+    };
+    const handleClose = () => {
+        setAnchorEl(null);
+    };
+
+    const handelEdit = () => {
+        const { group } = lesson;
+        const editObj = {
+            id: itemData.id,
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/EditCurrentSchedule/ScheduleBoard/ScheduleBoardItem.js]: @HiroFive maybe destructuring itemData

File: frontend/src/components/LessonsPage/LessonPage.js [hunk 7]
```diff
@@ -59,47 +64,39 @@ const LessonPage = (props) => {
...
 
     useEffect(() => {
-        showAllTeachersService();
-        getLessonTypes();
-        showAllGroupsService();
+        showAllTeachers();
+        getLessonTypesStart();
+        getEnabledGroupsStart();
         showAllSubjectsService();
         showAllSemestersService();
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/LessonsPage/LessonPage.js]: @Yashchemskyi-Oleksandr we have saga for showAllSemestersService()

**Review by ViktoriiaKosovan — APPROVED** [review]

File: frontend/src/components/EditCurrentSchedule/ScheduleBoard/ScheduleBoardItem.js (hunk 5 above)
Comment (HiroFive) [reply frontend/src/components/EditCurrentSchedule/ScheduleBoard/ScheduleBoardItem.js]: fixed

File: frontend/src/components/EditCurrentSchedule/ScheduleBoard/ScheduleBoardItem.js (hunk 6 above)
Comment (HiroFive) [reply frontend/src/components/EditCurrentSchedule/ScheduleBoard/ScheduleBoardItem.js]: done

File: frontend/src/components/LessonsPage/LessonPage.js (hunk 7 above)
Comment (HiroFive) [reply frontend/src/components/LessonsPage/LessonPage.js]: done

File: frontend/src/actions/schedule.js (hunk 3 above)
Comment (HiroFive) [reply frontend/src/actions/schedule.js]: changed

File: frontend/src/components/Semester/SemesterList/SemesterList.js (hunk 2 above)
Comment (ViktoriiaKosovan) [reply frontend/src/components/Semester/SemesterList/SemesterList.js]: its action

File: frontend/src/actions/schedule.js (hunk 4 above)
Comment (HiroFive) [reply frontend/src/actions/schedule.js]: changed

File: frontend/public/assets/i18n/translations/en/common.json [hunk 8]
```diff
@@ -181,5 +180,16 @@
     "hour_one": "<0>{{count}}</0> hour",
     "hour_other": "<0>{{count}}</0> hours",
     "move_label": "Move",
-    "more_label": "More"
+    "more_label": "More",
+    "select_file": "Select file",
+    "select-correct_format": "*File has to be in csv format",
+    "file-rules-for-each-line": "*Each line of the file should consist of four fields, separated by commas.",
+    "example-file": "Example of file",
+    "file-rules-for-header": "*First line of the file is a header.",
```
Comment (andrii-kuzmichov) [inline frontend/public/assets/i18n/translations/en/common.json]: snake case

File: frontend/public/assets/i18n/translations/en/common.json [hunk 9]
```diff
@@ -181,5 +180,16 @@
...
+    "table_columns_size": "Table columns size",
+    "columns_size_small": "small",
+    "columns_size_base": "base",
+    "columns_size_large": "large",
+    "search_params": "Search options for available rooms",
+    "free-rooms-list-empty": "The list of available rooms is empty"
```
Comment (andrii-kuzmichov) [inline frontend/public/assets/i18n/translations/en/common.json]: the same

File: frontend/public/assets/i18n/translations/uk/common.json [hunk 10]
```diff
@@ -179,5 +179,15 @@
...
+    "select-correct_format": "*Формат файлу повинен бути csv",
+    "file-rules-for-each-line": "*Кожний рядок повинен мати 4 поля через кому.",
+    "example-file": "Приклад файлу.",
+    "file-rules-for-header": "*Перший рядок файлу - це заголовок.",
+    "table_columns_size": "Розмір стовпців таблиці",
+    "columns_size_small": "малі",
+    "columns_size_base": "базові",
+    "columns_size_large": "великі",
+    "search_params": "Параметри пошуку вільних кімнати",
+    "free-rooms-list-empty": "Список вільних кімнат порожній"
```
Comment (andrii-kuzmichov) [inline frontend/public/assets/i18n/translations/uk/common.json]: the same

File: frontend/src/App.scss [hunk 11]
```diff
@@ -1,236 +1,196 @@
...
-}
-
-@mixin buttonStyle {
-  cursor: pointer;
-  width: $btn-size;
-  height: $btn-size;
-  transition: color 0.3s linear;
+html {
+    height: $full-size;
 }
```
Comment (andrii-kuzmichov) [inline frontend/src/App.scss:8]: why we need it?

File: frontend/src/App.scss [hunk 12]
```diff
@@ -1,236 +1,196 @@
...
-}
 
-.margin-top-bottom {
-  margin: 2px 0 7px 0;
+    .buttons-style {
+        width: 48%;
+
+        @media (max-width: $md-m) {
+            margin-bottom: 5px;
+            font-size: 12px;
```
Comment (andrii-kuzmichov) [inline frontend/src/App.scss]: add constants

**Review by andrii-kuzmichov — APPROVED** [review]

File: frontend/public/assets/i18n/translations/en/common.json (hunk 8 above)
Comment (ViktoriiaKosovan) [reply frontend/public/assets/i18n/translations/en/common.json]: done

File: frontend/public/assets/i18n/translations/en/common.json (hunk 9 above)
Comment (ViktoriiaKosovan) [reply frontend/public/assets/i18n/translations/en/common.json]: done

File: frontend/public/assets/i18n/translations/uk/common.json (hunk 10 above)
Comment (ViktoriiaKosovan) [reply frontend/public/assets/i18n/translations/uk/common.json]: done

File: frontend/src/App.scss (hunk 12 above)
Comment (ViktoriiaKosovan) [reply frontend/src/App.scss]: done

**Pull request merged**